demovieworderhooks/demovieworderhooks.php

```diff
@@ -0,0 +1,222 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+use PrestaShop\Module\DemoViewOrderHooks\Install\InstallerFactory;
+use PrestaShop\Module\DemoViewOrderHooks\Presenter\OrderLinkPresenter;
+use PrestaShop\Module\DemoViewOrderHooks\Presenter\OrderReviewPresenter;
+use PrestaShop\Module\DemoViewOrderHooks\Presenter\OrdersPresenter;
+use PrestaShop\Module\DemoViewOrderHooks\Presenter\PackageLocationsPresenter;
+use PrestaShop\Module\DemoViewOrderHooks\Presenter\SignaturePresenter;
+use PrestaShop\Module\DemoViewOrderHooks\Repository\OrderRepository;
+use PrestaShop\Module\DemoViewOrderHooks\Repository\OrderReviewRepository;
+use PrestaShop\Module\DemoViewOrderHooks\Repository\PackageLocationRepository;
+use PrestaShop\Module\DemoViewOrderHooks\Repository\SignatureRepository;
+
+class DemoViewOrderHooks extends Module
+{
+    public function __construct()
+    {
+        $this->name = 'demovieworderhooks';
+        $this->author = 'PrestaShop';
+        $this->version = '1.0.0';
+        $this->displayName = 'Demonstration of new hooks in PrestaShop 1.7.7 order view page';
+        $this->ps_versions_compliancy = ['min' => '1.7.7.0', 'max' => _PS_VERSION_];
+
+        parent::__construct();
+    }
+
+    public function install()
+    {
+        if (!parent::install()) {
+            return false;
+        }
+
+        $installer = InstallerFactory::create();
+
+        return $installer->install($this);
+    }
+
+    public function uninstall()
+    {
+        $installer = InstallerFactory::create();
+
+        return $installer->uninstall() && parent::uninstall();
+    }
+
+    /**
+     * Displays customer's signature.
+     */
+    public function hookDisplayBackOfficeOrderActions(array $params)
+    {
+        /** @var SignatureRepository $signatureRepository */
+        $signatureRepository = $this->get('prestashop.module.demovieworderhooks.repository.signature_repository');
+
+        /** @var SignaturePresenter $signaturePresenter */
+        $signaturePresenter = $this->get('prestashop.module.demovieworderhooks.presenter.signature_presenter');
+
+        $signature = $signatureRepository->findOneBy(['orderId' => $params['id_order']]);
+
+        if (!$signature) {
+            return '';
+        }
+
+        return $this->render($this->getModuleTemplatePath() . 'customer_signature.html.twig', [
+            'signature' => $signaturePresenter->present($signature, (int) $this->context->language->id),
+        ]);
+    }
+
+    public function hookDisplayAdminOrderContentOrder(array $params)
+    {
+        return 'displayAdminOrderContentOrder';
```

Comment:
This one is left with nothing in it. Shall we add something like a logo or an image here 😄 ?

Reply:
this one is deprecated so will delete it. See:     `**(deprecated since 1.7.7.0) - use displayAdminOrderTabLink and displayAdminOrderTabContent**`  See it in: https://github.com/PrestaShop/docs/pull/456/files#diff-d839e821025aa394f3fdce943cc3b0ca